Shared/Tests.Shared/AddOrUpdateTests.cs

```diff
@@ -846,6 +846,86 @@ public void AddOrUpdate_WhenObjectHasNonEmptyList_ShouldNotThrow()
             }, Throws.Nothing);
         }
 
+        [Test]
+        public void AddOrUpdate_WhenListIsNull_ShouldNotThrow()
+        {
+            var first = new PrimaryKeyWithPKList
+            {
+                Id = 42
+            };
+
+            Assert.That(() =>
+            {
+                _realm.Write(() => _realm.Add(first, update: true));
+            }, Throws.Nothing);
+        }
+
+        [Test]
+        public void AddOrUpdate_WhenListObjectsHavePK_ShouldOverwriteList()
+        {
+            var first = new PrimaryKeyWithPKList
+            {
+                Id = 1
+            };
+
+            first.ListValue.Add(new PrimaryKeyObject
+            {
+                Id = 1
+            });
+
+            _realm.Write(() => _realm.Add(first));
+
+            var second = new PrimaryKeyWithPKList
+            {
+                Id = 1
+            };
+
+            second.ListValue.Add(new PrimaryKeyObject
+            {
+                Id = 2
+            });
+
+            _realm.Write(() => _realm.Add(second, update: true));
+
+            Assert.That(first.ListValue, Is.EquivalentTo(second.ListValue));
+            Assert.That(first.ListValue.Count, Is.EqualTo(1));
+            Assert.That(first.ListValue[0].Id, Is.EqualTo(2));
+            Assert.That(_realm.All<PrimaryKeyObject>().Count(), Is.EqualTo(2));
```

Comment:
I get that, I just don't think it proves the point very well because of the confusion of where the **two** objects come from. At the very least, please repeat the assertion so it is before _and_after the `add(...update:true)` - **then** it will be proving the count didn't change.